accounts/geth/src/main/java/pm/gnosis/heimdall/accounts/repositories/impl/GethAccountsRepository.kt

```diff
@@ -73,4 +74,8 @@ class GethAccountsRepository @Inject constructor(
             Completable.fromCallable {
                 gethKeyStore.importECDSAKey(privateKey, generateRandomString())
             }
+
+    override fun generateMnemonic(): Single<String> = Single.just(bip39.generateMnemonic())
```

Comment:
This is happening on the UI thread, correct? Is that on purpose?

Reply:
The `ViewModel` specifies on which thread it will run. I think it makes more sense in this way because we can have situations where we already are on a different thread and we want this computation to be executed there